src/primitives/transaction.h

```diff
@@ -475,9 +593,13 @@ struct CMutableTransaction
     std::vector<CTxOut> vout;
     uint32_t nLockTime;
     uint32_t nExpiryHeight;
+    CAmount valueBalance;
+    std::vector<SpendDescription> vShieldedSpend;
+    std::vector<OutputDescription> vShieldedOutput;
     std::vector<JSDescription> vjoinsplit;
     uint256 joinSplitPubKey;
     CTransaction::joinsplit_sig_t joinSplitSig = {{0}};
+    CTransaction::binding_sig_t bindingSig = {{0}};
```

Comment:
Dislike the duplication of code between ``CTransaction`` and ``CMutableTransaction``, but that's a longer-term thing.

Reply:
Yep, I have plans :upside_down_face: